mesa/visualization/components/matplotlib.py

```diff
@@ -146,132 +180,269 @@ def draw_property_layers(ax, space, propertylayer_portrayal, model):
             )
 
 
-def _draw_grid(space, space_ax, agent_portrayal, propertylayer_portrayal, model):
-    if propertylayer_portrayal:
-        draw_property_layers(space_ax, space, propertylayer_portrayal, model)
+def collect_agent_data(
+    space: OrthogonalGrid | HexGrid | Network | ContinuousSpace,
+    agent_portrayal: Callable,
+    c_default="tab:blue",
```

Comment:
Two things here: 
For API stability I think it would be better if this were a dictionary named "defaults". Maybe we will in the future add more things that can be collected and then it wont grow into an endless list and also we don't have to worry about the order. Now its c, marker, size, whereas matplotlibs order is s, c, marker.

Secondly I would much prefer if we don't use the abbreviated names c and s, even if that is used by matplotlibs api. But we don't want our collect function be tight to matplotlib API, it should be usable for different drawing backends as well.

Reply:
> For API stability I think it would be better if this were a dictionary named "defaults". 
I was actually thinking of using a named tuple, rather than a dict, but I would need to take a closer look what works best. Note that since these are keyword arguments, order does not matter.

> Secondly I would much prefer if we don't use the abbreviated names c and s, even if that is used by matplotlibs api

I find this a bit tricky. The only other "backend" at the moment is altair, and none of this will work for altair plotting given how radically different altair is (for the record, I really like altair). I am, however, fine with reverting to the current labels (i.e., color, size, marker).

Reply:
If we really want to support multiple backends (which is definitely debatable), I think we need two sets of terms:
- Shared arguments, like "color" and "size" for example, that work with multiple (preferably all) backends. We translate them to the proper input value for that backend on the fly.
- Backend-specific arguments, that work only with that specific backend.

For shared argument, ideally those would be actual arguments, which can be type hinted and document, instead of dict keys.

Reply:
Maybe the term "backend" was a bit misleading in this context, given my other suggestion about a generic draw_space function with only matplotlib and altair possibilites. But in the context of this `collect_agent_data` function it just refers to the idea of collecting data specific to the plotting requirements. Its independent on how you use that data for plotting, but you need that data for any plotting library. 

Also note that its the exact requirement we have for (a new, possible) data collector: Iterating over an agentset and collecting some variables (here: c, s and marker), possibly transforming them on the way. Not sure what to make with that information 😆